w-dms/restHandlers.go

```diff
@@ -0,0 +1,235 @@
+// Package classification Task API.
+//
+// Provides API to operate with tasks
+//
+// swagger:meta
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+type swaggerTask struct {
+	//Task Id
+	ID string `json:"id" bson:"_id,omitempty"`
+	// Task Alias
+	// required: true
+	Alias string `json:"alias" bson:"alias"`
+	// Task description
+	Description string `json:"desc"`
+	// Task type
+	// required: true
+	Type string `json:"type"`
+	// Task tags
+	// required: false
+	Tags []string `json:"tags"`
+	// Timastamp when the task was created
+	Timestamp int64 `json:"ts"`
+	// Task estimation time
+	EstimateTime string `json:"etime"`
+	// Task actual time
+	RealTime string `json:"rtime"`
+	// Times when the task should be reminded
+	Reminders []string `json:"reminders"`
+}
+
+// Tasks were read successfully
+// swagger:response taskResponse
+type swaggerTaskResponse struct {
+	//in: body
+	Body []swaggerTask
+}
+
+// Query params
+//swagger:parameters readTasks deleteTask
+type swaggerReadTaskParameters struct {
+	// Id of the task to be found (skip to find all)
+	//
+	//in: query
+	//required: false
+	ID string `json:"id"`
+
+	// Alias of the task to be found (skip to find all)
+	// in: query
+	//required: false
+	Alias string `json:"alias"`
+}
+
+// Task
+//swagger:parameters saveTask updateTask
+type swaggerTaskRequest struct {
+	//in: body
+	Body swaggerTask
+}
+
+// Task was not found
+//swagger:response taskNotFound
+type swaggerTaskNotFound struct{}
+
+// Task created
+//swagger:response taskCreated
+type swaggerTaskCreated struct{}
+
+// Task updated
+//swagger:response taskUpdated
+type swaggerTaskUpdated struct{}
+
+// Task deleted
+//swagger:response taskDeleted
+type swaggerTaskDeleted struct{}
+
+func closeCompletelyRequestBody(body *io.ReadCloser) {
+	defer func() {
+		err := (*body).Close()
+		if err != nil {
+			log.Println(err.Error())
+		}
+	}()
+	_, err := ioutil.ReadAll(*body)
+	if err != nil {
+		log.Println(err.Error())
+	}
+}
+
+func readTaskFromRequest(request *http.Request) (*Task, error) {
+	defer closeCompletelyRequestBody(&request.Body)
```

Comment:
`request.Body` drain is not needed on server.
Also, there's no need to use pointers to interfaces as they are reference types.